### PR TITLE
add unit tests for template transform node

### PR DESCRIPTION
**Pull request by 0xsatoshi99**
Added unit tests for the template_transform workflow node.

Fixes #28700

Tests cover:
- Entity validation (TemplateTransformNodeData)
- Node initialization and configuration
- Simple and complex Jinja2 template transformations
- Error handling (code execution errors, output length limits)
- Variable handling (None values, numeric, dict, list types)
- Edge cases (empty variables, static templates)

*Contribution by Gittensor, learn more at https://gittensor.io/*

### CODE REVIEW

File: api/tests/unit_tests/core/workflow/nodes/template_transform/template_transform_node_spec.py [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+                tenant_id="test_tenant",
+                app_id="test_app",
+                workflow_type=WorkflowType.WORKFLOW,
+                workflow_id="test_workflow",
+                graph_config={},
+                user_id="test_user",
+                user_from="test",
+                invoke_from="test",
+                call_depth=0,
+            ),
```
[the remark this answers is not in the document]
Comment (0xsatoshi99) [reply api/tests/unit_tests/core/workflow/nodes/template_transform/template_transform_node_spec.py]: fixed

**Comment by 0xsatoshi99** [comment]
@crazywoola Can you plz check this PR?

**Comment by crazywoola** [comment]
Please link an issue or create a new one first :)

**Comment by 0xsatoshi99** [comment]
@crazywoola new issue created: https://github.com/langgenius/dify/issues/28700, and connected this pr to that issue. And I fixed linting issue, please check again.

**Review by crazywoola — DISMISSED** [review]

**Review by crazywoola — APPROVED** [review]
